app/controllers/jobs_controller.rb

```diff
@@ -43,7 +43,7 @@ def show
     if params[:flavor] == 'failed'
       job = Delayed::Job::Failed.find(params[:id])
     else
-      job = Delayed::Job.find(params[:id])
+      job = Delayed::Job::Failed.find(params[:id]) rescue Delayed::Job.find(params[:id])
```

Comment:
Thanks for the pointer on `.wrap`. Good point on the potential ID collision. I think it's best if I just revert out that change and keep the original behaviour.